### ISSUE TITLE
Prediction Probability varies with each run

### ISSUE THREAD

**Issue by Benuraj** [facebook/C3D#114]
Hi I was able to train the system using this repo for 3D convolution but I'm currently facing a weird issue. My prediction probability is non-deterministic i.e., at every run i get different different probability values.

**Comment by dutran** [comment]
Can you post your testing command as well as prototxt file?

**Comment by dutran** [comment]
@Benuraj are you good now?

**Comment by Benuraj** [comment]
hey @dutran sorry for the delayed response.. My test prototxt is following

name: "deep_c3d_test"
layers {
  name: "data"
  type: VIDEO_DATA
  top: "data"
  top: "label"
  image_data_param {
    source: "/PathToData/C3D/examples/c3d_finetuning/test_01.lst"
    use_image: true
    mean_file: "/PathToMean/lmean.binaryproto"
    batch_size: 1
    mirror: false
    crop_size : 0
    show_data: 0
    new_height: 128
    new_width: 171
    new_length: 16
    shuffle: false
  }
}
# ----------- 1st layer conv ---------------
layers {
  name: "conv1a"
  type: CONVOLUTION3D
  bottom: "data"
  top: "conv1a"
  blobs_lr: 1
  blobs_lr: 2
  weight_decay: 1
  weight_decay: 0
  convolution_param {
    num_output: 8
    kernel_size: 3
    kernel_depth: 3
    pad: 1
    temporal_pad: 1
    stride: 1
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layers {
  name: "relu1a"
  type: RELU
  bottom: "conv1a"
  top: "conv1a"
}

# ------------- 2nd layer conv --------------
layers {
  name: "conv2a"
  type: CONVOLUTION3D
  bottom: "conv1a"
  top: "conv2a"
  blobs_lr: 1
  blobs_lr: 2
  weight_decay: 1
  weight_decay: 0
  convolution_param {
    num_output: 16
    kernel_size: 3
    kernel_depth: 3
    pad: 0
    temporal_pad: 0
    stride: 1
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 1
    }
  }
}
layers {
  name: "relu2a"
  type: RELU
  bottom: "conv2a"
  top: "conv2a"
}

# ----------------- 3rd conv layer --------------
layers {
  name: "conv3a"
  type: CONVOLUTION3D
  bottom: "conv2a"
  top: "conv3a"
  blobs_lr: 1
  blobs_lr: 2
  weight_decay: 1
  weight_decay: 0
  convolution_param {
    num_output: 16
    kernel_size: 3
    kernel_depth: 3
    pad: 0
    temporal_pad: 0
    stride: 1
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 1
    }
  }
}
layers {
  name: "relu3a"
  type: RELU
  bottom: "conv3a"
  top: "conv3a"
}
layers {
  name: "pool3"
  type: POOLING3D
  bottom: "conv3a"
  top: "pool3"
  pooling_param {
    pool: MAX
    kernel_size: 2
    kernel_depth: 2
    stride: 2
    temporal_stride: 2
  }
}

# --------- 4th conv layer --------
layers {
  name: "conv4a"
  type: CONVOLUTION3D
  bottom: "pool3"
  top: "conv4a"
  blobs_lr: 1
  blobs_lr: 2
  weight_decay: 1
  weight_decay: 0
  convolution_param {
    num_output: 32
    kernel_size: 3
    kernel_depth: 1
    pad: 0
    temporal_pad: 0
    stride: 1
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 1
    }
  }
}
layers {
  name: "relu4a"
  type: RELU
  bottom: "conv4a"
  top: "conv4a"
}
layers {
  name: "pool4"
  type: POOLING3D
  bottom: "conv4a"
  top: "pool4"
  pooling_param {
    pool: MAX
    kernel_size: 2
    kernel_depth: 1
    stride: 2
    temporal_stride: 1
  }
}

# --------------- 5th conv layer --------
layers {
  name: "conv5a"
  type: CONVOLUTION3D
  bottom: "pool4"
  top: "conv5a"
  blobs_lr: 1
  blobs_lr: 2
  weight_decay: 1
  weight_decay: 0
  convolution_param {
    num_output: 64
    kernel_size: 3
    kernel_depth: 1
    pad: 0
    temporal_pad: 0
    stride: 1
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 1
    }
  }
}
layers {
  name: "relu5a"
  type: RELU
  bottom: "conv5a"
  top: "conv5a"
}

# --------------- 6th conv layer --------
layers {
  name: "conv6a"
  type: CONVOLUTION3D
  bottom: "conv5a"
  top: "conv6a"
  blobs_lr: 1
  blobs_lr: 2
  weight_decay: 1
  weight_decay: 0
  convolution_param {
    num_output: 128
    kernel_size: 3
    kernel_depth: 1
    pad: 0
    temporal_pad: 0
    stride: 1
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 1
    }
  }
}
layers {
  name: "relu6a"
  type: RELU
  bottom: "conv6a"
  top: "conv6a"
}

# --------------- 7th conv layer --------
layers {
  name: "conv7a"
  type: CONVOLUTION3D
  bottom: "conv6a"
  top: "conv7a"
  blobs_lr: 1
  blobs_lr: 2
  weight_decay: 1
  weight_decay: 0
  convolution_param {
    num_output: 256
    kernel_size: 3
    kernel_depth: 1
    pad: 0
    temporal_pad: 0
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 1
    }
  }
}
layers {
  name: "relu7a"
  type: RELU
  bottom: "conv7a"
  top: "conv7a"
}

layers {
  name: "pool5"
  type: POOLING3D
  bottom: "conv7a"
  top: "pool5"
  pooling_param {
    pool: MAX
    kernel_size: 2
    kernel_depth: 1
    stride: 2
    temporal_stride: 1
  }
}

# ---------------- fc layers -------------
layers {
  name: "fc8"
  type: INNER_PRODUCT
  bottom: "pool5"
  top: "fc8"
  blobs_lr: 1
  blobs_lr: 2
  weight_decay: 1
  weight_decay: 0
  inner_product_param {
    num_output: 2
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layers {
  name: "prob"
  type: SOFTMAX
  bottom: "fc8"
  top: "prob"
}
#layers {
 # top: "accuracy"
 # name: "accuracy"
 # type: ACCURACY
 # bottom: "prob"
 # bottom: "label"
#}

**Comment by dutran** [comment]
crop_size 0, mean you input empty frames. It is supposed to be 112.
